tinygrad/engine/schedule.py

```diff
@@ -32,15 +32,15 @@
   (UPat(GroupOp.Movement, name="mv", src=(UPat.var("x"),)), lambda mv,x:
    # naturally correct
    (isinstance(mv.arg, tuple) and mv.dtype == x.dtype) or
-   # TODO: "make things that can't be images not images" can override the source dtype
-   # is there a clean way to update its _mop children?
+   # "make things that can't be images not images" can change the buffer dtype
+   # this is fine as long as it's a realized buffer and base dtypes match.
    ((isinstance(mv.dtype, ImageDType) or isinstance(x.dtype, ImageDType)) and x.dtype.base == mv.dtype.base and x.is_realized)),
 
   # Tensor variable bindings
   (UPat(Ops.BIND, dtypes.int, (UPat(Ops.DEFINE_VAR), UPat.cvar(dtype=dtypes.int)), arg=None), lambda: True),
 
   # Tensor const has a ShapeTracker of shape=() and a device
-  (UPat(Ops.CONST, src=(UPat(Ops.VIEW, src=(UPat(Ops.DEVICE),)),)), lambda: True),
+  (UPat(Ops.CONST, src=(UPat(Ops.VIEW, name="st", src=(UPat(Ops.DEVICE),)),)), lambda st: all(v.mask is None for v in st.st.views)),
```

Comment:
this reads weird to me as if you are checking already, it should only has one view and that view has all 0 stride.

checking all mask is None in (multiple) views has very little semantic meaning

Reply:
is there a better way to check if the st is unmasked?

Reply:
if you want to a check is a st is unmasked, maybe no None in real_strides, but i don't think that's what you want. also not clear why checking mask is None for all views is right. won't it contain false negatives?

const shapetracker should be single view, no mask, and all 0 stride, or there's something wrong with view math